app/Services/OrderService.php

```diff
@@ -0,0 +1,106 @@
+<?php
+
+namespace App\Services;
+
+use App\Jobs\SoftDeleteOrder;
+use App\Models\Order;
+use App\Models\Product;
+use App\Models\Product_order;
+use Exception;
+use Illuminate\Support\Facades\DB;
+
+class OrderService
+{
+
+    public static function store($data, $products_id_quantity){
+
+        DB::beginTransaction();
+        try{
+            $order = Order::create($data);
+
+            foreach ($products_id_quantity as $product_id_quantity){
+                $order->products()->attach($product_id_quantity['product_id']
+                    , ['quantity'=> $product_id_quantity['quantity']]);
+            }
+
+            if(!OrderService::subtractStock($order)){
+                DB::rollback();
+                return response()->json(['Not enough quantity of product available'], 400);
```

Comment:
do not return a response on services.
return message, and handle it on controller